src/api/app/controllers/webui/monitor_controller.rb

```diff
@@ -126,6 +119,23 @@ def discard_cache?
 
   private
 
+  def get_history(prefix, arch_name, range)
+    gethistory("#{prefix}_#{arch_name}", range, !discard_cache?)
+  end
+
+  def gethistory(key, range, cache = 1)
+    cachekey = "#{key}-#{range}"
+    Rails.cache.delete(cachekey, shared: true) unless cache
+    Rails.cache.fetch(cachekey, expires_in: (range.to_i * 3600) / 150, shared: true) do
+      hash = StatusHistory.history_by_key_and_hours(key, range)
+      hash.sort_by { |a| a[0] }
+    end
+  end
+
+  def set_default_arch
```

Comment:
[nitpick]

It's default arch or default architecture? I would follow the same naming.